tidb-external-ts.md

```diff
@@ -0,0 +1,122 @@
+
+## Feature description
+
+The system variable [`tidb_external_ts`](/system-variables.md#tidb_external_ts-new-in-v640) specifies the timestamp of the historical data to be read when `tidb_enable_external_ts_read` is enabled.
+
+The system variable [`tidb_enable_external_ts_read`](/system-variables.md#tidb_enable_external_ts_read-new-in-v640) controls whether to read historical data in the current session or globally. The default value is `OFF`, which means the reading historical data function is disabled, and the `tidb_external_ts` value is ignored. When this variable is set to `ON` globally, all queries read historical data before the time specified by `tidb_external_ts`. If `tidb_enable_external_ts_read` is set to `ON` only for a certain session, only queries in that session read historical data.
+
+When the `tidb_enable_external_ts_read` is enabled, TiDB becomes read-only. All write queries will fail with an error like `ERROR 1836 (HY000): Running in read-only mode`.
+
+## Usage examples
+
+This section describes how to use the `tidb_external_ts` variable to read historical data with examples.
+
+1. Create a table and insert some rows into the table:
+
+    ```sql
+    CREATE TABLE t (c INT);
+    ```
+
+    ```
+    Query OK, 0 rows affected (0.01 sec)
+    ```
+
+    ```sql
+    INSERT INTO t VALUES (1), (2), (3);
+    ```
+
+    ```
+    Query OK, 3 rows affected (0.00 sec)
+    ```
+
+2. View the data in the table:
+
+    ```sql
+    SELECT * FROM t;
+    ```
+
+    ```
+    +------+
+    | c    |
+    +------+
+    |    1 |
+    |    2 |
+    |    3 |
+    +------+
+    3 rows in set (0.00 sec)
+    ```
+
+3. Set the `tidb_external_ts` to `@@tidb_current_ts`:
+
+    ```sql
+    START TRANSACTION;
+    SET GLOBAL tidb_external_ts=@@tidb_current_ts;
+    COMMIT;
+    ```
+
+4. Insert a new row and confirm that it is inserted:
+
+    ```sql
+    INSERT INTO t VALUES (4);
+    ```
+
+    ```
+    Query OK, 1 row affected (0.001 sec)
+    ```
+
+    ```sql
+    SELECT * FROM t;
+    ```
+
+    ```
+    +------+
+    | id   |
+    +------+
+    |    1 |
+    |    2 |
+    |    3 |
+    |    4 |
+    +------+
+    4 rows in set (0.00 sec)
+    ```
+
+5. Set the `tidb_enable_external_ts_read` to `ON` and then view data in the table:
+
+    ```sql
+    SET tidb_enable_external_ts_read=ON;
+    SELECT * FROM t;
+    ```
+
+    ```
+    +------+
+    | c    |
+    +------+
+    |    1 |
+    |    2 |
+    |    3 |
+    +------+
+    3 rows in set (0.00 sec)
+    ```
+
+    As the `tidb_external_ts` is set to the timestamp before the new row is inserted, the newly inserted row is not returned after the `tidb_enable_external_ts_read` is enabled.
```

Comment:
```suggestion
    Because `tidb_external_ts` is set to the timestamp before the new row is inserted, the newly inserted row is not returned after the `tidb_enable_external_ts_read` is enabled.
```